feg/gateway/services/n7_proxy/metrics/metrics.go

```diff
@@ -0,0 +1,229 @@
+	)
+}
+
+func NewSessionHealthTracker() *SessionHealthTracker {
+	initMetrics := &SessionHealthMetrics{
+		SmPolicyCreateTotal:    0,
+		SmpolicyCreateFailures: 0,
+		SmPolicyUpdateTotal:    0,
+		SmPolicyUpdateFailures: 0,
+		SmPolicyDeleteTotal:    0,
+		SmPolicyDeleteFailures: 0,
+		N7Timeouts:             0,
+	}
+	defaultMetrics := &SessionHealthTracker{
+		Metrics:                 initMetrics,
+		RequestFailureThreshold: float32(DefaultRequestFailureThresholdPct),
+		MinimumRequestThreshold: uint32(DefaultMinimumRequestThreshold),
+	}
+	configPtr := &mcfgprotos.N7Config{}
+	err := mconfig.GetServiceConfigs("n7_proxy", configPtr)
+	if err != nil {
+		return defaultMetrics
+	}
+	reqFailureThreshold := configPtr.GetRequestFailureThreshold()
+	minReqThreshold := configPtr.GetMinimumRequestThreshold()
+	if reqFailureThreshold == 0 {
+		glog.Info("Request failure threshold cannot be 0; Using default health parameters...")
+		return defaultMetrics
+	}
+	if minReqThreshold == 0 {
+		glog.Info("Minimum request threshold cannot be 0; Using default health parameters...")
+		return defaultMetrics
+	}
+	return &SessionHealthTracker{
+		Metrics:                 initMetrics,
+		RequestFailureThreshold: reqFailureThreshold,
+		MinimumRequestThreshold: minReqThreshold,
+	}
+}
+
+func GetCurrentHealthMetrics() (*SessionHealthMetrics, error) {
+	smPolicyCreateTotal, err := service_health_metrics.GetInt64("pcf_sm_policy_create_requests_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyCreateFailures, err := service_health_metrics.GetInt64("pcf_sm_policy_create_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyUpdateTotal, err := service_health_metrics.GetInt64("pcf_sm_policy_update_requests_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyUpdateFailures, err := service_health_metrics.GetInt64("pcf_sm_policy_update_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyDeleteTotal, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_requests_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyDeleteFailure, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	n7Timeouts, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	return &SessionHealthMetrics{
+		SmPolicyCreateTotal:    smPolicyCreateTotal,
+		SmpolicyCreateFailures: smPolicyCreateFailures,
+		SmPolicyUpdateTotal:    smPolicyUpdateTotal,
+		SmPolicyUpdateFailures: smPolicyUpdateFailures,
+		SmPolicyDeleteTotal:    smPolicyDeleteTotal,
+		SmPolicyDeleteFailures: smPolicyDeleteFailure,
+		N7Timeouts:             n7Timeouts,
+	}, nil
+}
+
+func (prevMetrics *SessionHealthMetrics) GetDelta(
+	currentMetrics *SessionHealthMetrics,
+) (*SessionHealthMetrics, error) {
+	if currentMetrics == nil {
+		return nil, fmt.Errorf("nil current health metrics provided")
```

Comment:
What about 
```
GetDelta has nil SessionHealthMetrics
```
This way it gives more information about the call and  it is easy to find the error doing a quick search